src/symbol/projection.js

```diff
@@ -276,11 +290,19 @@ function placeGlyphsAlongLine(symbol,
     return {};
 }
 
+function projectTruncatedLineSegment(previousTilePoint: Point, currentTilePoint: Point, previousProjectedPoint: Point, minimumLength: number, projectionMatrix: mat4) {
+    const projectedUnitVertex = project(previousTilePoint.add(previousTilePoint.sub(currentTilePoint)._unit()), projectionMatrix).point;
```

Comment:
🤔  That's a good question. I think the answer is "not under any reasonable conditions," but let me try to reason why:

 - Any point that's near the plane of the camera is very far "away" (in x/y terms) from the viewport
 - We're only placing glyphs whose anchor is within the viewport
 - For a label to reach from within the viewport to a line vertex at the edge of camera plane, it would need to be _very_ long (much longer than could be shown on one screen)
 - A label with `text-pitch-alignment: map` would be the best candidate for being way longer than expected, because it would get stretched out more and more the closer it got to the plane of the camera. But that case wouldn't be affected by the "plane of the camera" logic, because it would be projected using a tile-based matrix (see logic in `getLabelPlaneMatrix`).
 - Therefore any label we're placing should fit on the line _before_ the hypothetical vertex that's right at the edge of the camera plane, and we should never reach a point where we try to project past the hypothetical vertex.

I guess another way of thinking about it is: if we needed to draw a glyph that was on a line just next to the plane of the camera, we'd be very close to needing to draw a glyph that was just behind the plane of the camera, and we'd be hosed either way.